### ISSUE TITLE
Feature request: log Scratch Link communication for debugging purposes

### ISSUE THREAD

**Issue by cwillisf** [LLK/scratch-link#122]
Scratch Link could log peripheral communication to a file and/or a semi-hidden debug dialog as a way to help with extension development and debugging. Such a log should include fine-grained timestamps to help with rate limiting, queueing, etc.

Open question: should the log record WebSocket communication, hardware communication (like Bluetooth API calls), or both? One option would be to offer checkboxes so the user can enable or disable each as desired.

**Comment by knandersen** [comment]
Yeah having it be checkboxes would be super-sweet. I think both WebSocket comm. and hardware comm. are relevant!

**Comment by whaleygeek** [comment]
Lack of any debug at the coal-face makes this impossible to debug, and all we get is 'it doesn't work'.

We need both app logging and device logging, for diagnosis in the field. I have a school using micro:bits and this doesn't work at all (doesn't list any micro:bits). School meets all minimum requirements and firewall setup as documented here: https://support.microbit.org/support/solutions/articles/19000026080-using-the-micro-bit-with-scratch

Are there any events written to the windows application event log already that we could use to diagnose the cause, or anything useful in the browser console.log?

Just having a checkbox saying 'turn on diagnostics' and an entry in the scratch link that says 'find log file' would be great; a school could then email the detailed log to micro:bit support and it would give them half a chance to help the school fix the problem. The school would not need to understand the log, they would just email it to someone who can make sense of it. Textual format would be great as there are those of us that know how to trawl log files for useful info.

Also note the severely locked down school environment means the log file would need to be written to the registered user home directory whatever that is, as other usual locations won't have write permissions.

Thanks.
